src/rust/engine/fs/src/directory.rs

```diff
@@ -621,7 +621,20 @@ impl DigestTrie {
   ) {
     for entry in &*self.0 {
       let path = path_so_far.join(entry.name().as_ref());
-      f(&path, entry);
+
+      match (symlink_behavior.clone(), entry) {
```

Comment:
No need for the `.clone`, just have the enum derive `Copy`. Or just match on a reference here, i.e., `match (&symlink_behavior, entry)`, and you still use the existing match arm as-is because Rust will infer a dereference  as necessary to make the match arm work. See docs on "match binding modes", https://doc.rust-lang.org/stable/book/ch18-01-all-the-places-for-patterns.html.